### ISSUE TITLE
Error status code 400 with a specific coin

### ISSUE THREAD

**Issue by kamigreed** [Thornsnake/MCRBot#16]
I encountered the above error when the bot is checking the price of CVX. Somehow I own some of CVX tokens in my crypto exchange account when I can't even find it in my spot account.

I have tried reinstalling the bot, deleting the data folder and adding CVX to the ban list but nothing works. Any advice would be greatly appreciated!!!

**Comment by Thornsnake** [comment]
The prices are requested via the order book and both requests for USDT and USDC seem to work fine for me.
https://api.crypto.com/v2/public/get-book?instrument_name=CVX_USDT&depth=150
https://api.crypto.com/v2/public/get-book?instrument_name=CVX_USDC&depth=150

Has this error fixed itself, because I'm not able to reproduce it.
Is there a line number shown in the logs for this error message?

**Comment by kamigreed** [comment]
Do you mean the logs from doing -pm2 logs --lines 1000? Basically every 5 mins it will give me this:
2022-03-22T07:30:03: Error - Request failed with status code 400

Somehow, the coins I added to my ban list, like DOGE, USDC are also still being bought despite adding them to the ban list*.

*I've also just checked, I set it to buy at top 50 coins but the bot buys 60+ coins. I think somehow it's still reading my old configuration despite reinstalling and changing the values in the config file.

**Comment by kamigreed** [comment]
This may not help but a while back, I updated pm2 and npm. Perhaps it was something here that caused this error?

**Comment by Thornsnake** [comment]
The error message is probably not originating from the price check, but from the bot attempting to create an order. Error code 400 means that the order probably has been rejected because some parameters that were given to that order are invalid.

I have updated the code to produce more detailed error messages. Please update the bot and then check the error again that you are getting. There should be more info now.

**Comment by Thornsnake** [comment]
Regarding DOGE and USDC still being bought, that's probably because of crypto dust that was still in your portfolio. It should be sold automatically as soon as the REMOVAL time in your config.ts file is reached. If that is not the case, then stop the bot, delete the `/data/CoinRemovalList.json` file, manually sell the DOGE and USDC you have and then restart the bot.

**Comment by kamigreed** [comment]
Just updated the bot. For the past 15 mins, so far there's no errors. I'll monitor for 1-2 days and update here again when I get the error.

**Comment by kamigreed** [comment]
root/.pm2/logs/pm2-logrotate-error.log last 1000 lines:
/root/.pm2/logs/pm2-logrotate-out.log last 1000 lines:
1|pm2-logr | "/root/.pm2/logs/pm2-logrotate-out__2022-03-24_00-00-00.log" has been created
1|pm2-logr | "/root/.pm2/logs/bot-out__2022-03-24_00-00-00.log" has been created
1|pm2-logr | "/root/.pm2/logs/bot-error__2022-03-24_00-00-00.log" has been created
1|pm2-logr | "bot-out__2022-03-24_00-00-00.log" has been deleted
1|pm2-logr | "bot-error__2022-03-24_00-00-00.log" has been deleted
1|pm2-logr | "pm2-logrotate-out__2022-02-22_00-00-00.log" has been deleted

/root/.pm2/logs/bot-error.log last 1000 lines:
2|bot      | 2022-03-24T00:03:22: Error: Request failed with status code 400
2|bot      | 2022-03-24T00:03:22:     at createError (/root/MCRBot/node_modules/axios/lib/core/createError.js:16:15)
2|bot      | 2022-03-24T00:03:22:     at settle (/root/MCRBot/node_modules/axios/lib/core/settle.js:17:12)
2|bot      | 2022-03-24T00:03:22:     at IncomingMessage.handleStreamEnd (/root/MCRBot/node_modules/axios/lib/adapters/http.js:269:11)
2|bot      | 2022-03-24T00:03:22:     at IncomingMessage.emit (events.js:412:35)
2|bot      | 2022-03-24T00:03:22:     at endReadableNT (internal/streams/readable.js:1334:12)
2|bot      | 2022-03-24T00:03:22:     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
2|bot      | 2022-03-24T00:03:22:   config: {
2|bot      | 2022-03-24T00:03:22:     url: 'https://api.crypto.com/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:     method: 'post',
2|bot      | 2022-03-24T00:03:22:     data: '{"id":2,"method":"private/create-order","params":{"instrument_name":"CVX_USDT","side":"BUY","type":"MARKET","notional":0.119,"client_oid":"mcrbot_invest"},"nonce":1648080201965,"api_key":"MLNpqpT7yRoUihuSCZTdrt","sig":"3f2bc5ee75ab5291c56e4bfa4db31497d4b6115de277f4c3c67f61ee94e5a8ec"}',
2|bot      | 2022-03-24T00:03:22:     headers: {
2|bot      | 2022-03-24T00:03:22:       Accept: 'application/json, text/plain, */*',
2|bot      | 2022-03-24T00:03:22:       'Content-Type': 'application/json',
2|bot      | 2022-03-24T00:03:22:       'User-Agent': 'axios/0.21.2',
2|bot      | 2022-03-24T00:03:22:       'Content-Length': 285
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     transformRequest: [ [Function: transformRequest] ],
2|bot      | 2022-03-24T00:03:22:     transformResponse: [ [Function: transformResponse] ],
2|bot      | 2022-03-24T00:03:22:     timeout: 30000,
2|bot      | 2022-03-24T00:03:22:     adapter: [Function: httpAdapter],
2|bot      | 2022-03-24T00:03:22:     xsrfCookieName: 'XSRF-TOKEN',
2|bot      | 2022-03-24T00:03:22:     xsrfHeaderName: 'X-XSRF-TOKEN',
2|bot      | 2022-03-24T00:03:22:     maxContentLength: -1,
2|bot      | 2022-03-24T00:03:22:     maxBodyLength: -1,
2|bot      | 2022-03-24T00:03:22:     validateStatus: [Function: validateStatus],
2|bot      | 2022-03-24T00:03:22:     transitional: {
2|bot      | 2022-03-24T00:03:22:       silentJSONParsing: true,
2|bot      | 2022-03-24T00:03:22:       forcedJSONParsing: true,
2|bot      | 2022-03-24T00:03:22:       clarifyTimeoutError: false
2|bot      | 2022-03-24T00:03:22:     }
2|bot      | 2022-03-24T00:03:22:   },
2|bot      | 2022-03-24T00:03:22:   request: <ref *1> ClientRequest {
2|bot      | 2022-03-24T00:03:22:     _events: [Object: null prototype] {
2|bot      | 2022-03-24T00:03:22:       abort: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       aborted: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       connect: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       error: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       socket: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       timeout: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       prefinish: [Function: requestOnPrefinish]
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     _eventsCount: 7,
2|bot      | 2022-03-24T00:03:22:     _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:     outputData: [],
2|bot      | 2022-03-24T00:03:22:     outputSize: 0,
2|bot      | 2022-03-24T00:03:22:     writable: true,
2|bot      | 2022-03-24T00:03:22:     destroyed: false,
2|bot      | 2022-03-24T00:03:22:     _last: true,
2|bot      | 2022-03-24T00:03:22:     chunkedEncoding: false,
2|bot      | 2022-03-24T00:03:22:     shouldKeepAlive: false,
2|bot      | 2022-03-24T00:03:22:     _defaultKeepAlive: true,
2|bot      | 2022-03-24T00:03:22:     useChunkedEncodingByDefault: true,
2|bot      | 2022-03-24T00:03:22:     sendDate: false,
2|bot      | 2022-03-24T00:03:22:     _removedConnection: false,
2|bot      | 2022-03-24T00:03:22:     _removedContLen: false,
2|bot      | 2022-03-24T00:03:22:     _removedTE: false,
2|bot      | 2022-03-24T00:03:22:     _contentLength: null,
2|bot      | 2022-03-24T00:03:22:     _hasBody: true,
2|bot      | 2022-03-24T00:03:22:     _trailer: '',
2|bot      | 2022-03-24T00:03:22:     finished: true,
2|bot      | 2022-03-24T00:03:22:     _headerSent: true,
2|bot      | 2022-03-24T00:03:22:     socket: TLSSocket {
2|bot      | 2022-03-24T00:03:22:       _tlsOptions: [Object],
2|bot      | 2022-03-24T00:03:22:       _secureEstablished: true,
2|bot      | 2022-03-24T00:03:22:       _securePending: false,
2|bot      | 2022-03-24T00:03:22:       _newSessionPending: false,
2|bot      | 2022-03-24T00:03:22:       _controlReleased: true,
2|bot      | 2022-03-24T00:03:22:       secureConnecting: false,
2|bot      | 2022-03-24T00:03:22:       _SNICallback: null,
2|bot      | 2022-03-24T00:03:22:       servername: 'api.crypto.com',
2|bot      | 2022-03-24T00:03:22:       alpnProtocol: false,
2|bot      | 2022-03-24T00:03:22:       authorized: true,
2|bot      | 2022-03-24T00:03:22:       authorizationError: null,
2|bot      | 2022-03-24T00:03:22:       encrypted: true,
2|bot      | 2022-03-24T00:03:22:       _events: [Object: null prototype],
2|bot      | 2022-03-24T00:03:22:       _eventsCount: 10,
2|bot      | 2022-03-24T00:03:22:       connecting: false,
2|bot      | 2022-03-24T00:03:22:       _hadError: false,
2|bot      | 2022-03-24T00:03:22:       _parent: null,
2|bot      | 2022-03-24T00:03:22:       _host: 'api.crypto.com',
2|bot      | 2022-03-24T00:03:22:       _readableState: [ReadableState],
2|bot      | 2022-03-24T00:03:22:       _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:       _writableState: [WritableState],
2|bot      | 2022-03-24T00:03:22:       allowHalfOpen: false,
2|bot      | 2022-03-24T00:03:22:       _sockname: null,
2|bot      | 2022-03-24T00:03:22:       _pendingData: null,
2|bot      | 2022-03-24T00:03:22:       _pendingEncoding: '',
2|bot      | 2022-03-24T00:03:22:       server: undefined,
2|bot      | 2022-03-24T00:03:22:       _server: null,
2|bot      | 2022-03-24T00:03:22:       ssl: [TLSWrap],
2|bot      | 2022-03-24T00:03:22:       _requestCert: true,
2|bot      | 2022-03-24T00:03:22:       _rejectUnauthorized: true,
2|bot      | 2022-03-24T00:03:22:       parser: null,
2|bot      | 2022-03-24T00:03:22:       _httpMessage: [Circular *1],
2|bot      | 2022-03-24T00:03:22:       timeout: 30000,
2|bot      | 2022-03-24T00:03:22:       [Symbol(res)]: [TLSWrap],
2|bot      | 2022-03-24T00:03:22:       [Symbol(verified)]: true,
2|bot      | 2022-03-24T00:03:22:       [Symbol(pendingSession)]: null,
2|bot      | 2022-03-24T00:03:22:       [Symbol(async_id_symbol)]: 1527716,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kHandle)]: [TLSWrap],
2|bot      | 2022-03-24T00:03:22:       [Symbol(kSetNoDelay)]: false,
2|bot      | 2022-03-24T00:03:22:       [Symbol(lastWriteQueueSize)]: 0,
2|bot      | 2022-03-24T00:03:22:       [Symbol(timeout)]: Timeout {
2|bot      | 2022-03-24T00:03:22:         _idleTimeout: 30000,
2|bot      | 2022-03-24T00:03:22:         _idlePrev: [TimersList],
2|bot      | 2022-03-24T00:03:22:         _idleNext: [TimersList],
2|bot      | 2022-03-24T00:03:22:         _idleStart: 86026732,
2|bot      | 2022-03-24T00:03:22:         _onTimeout: [Function: bound ],
2|bot      | 2022-03-24T00:03:22:         _timerArgs: undefined,
2|bot      | 2022-03-24T00:03:22:         _repeat: null,
2|bot      | 2022-03-24T00:03:22:         _destroyed: false,
2|bot      | 2022-03-24T00:03:22:         [Symbol(refed)]: false,
2|bot      | 2022-03-24T00:03:22:         [Symbol(kHasPrimitive)]: false,
2|bot      | 2022-03-24T00:03:22:         [Symbol(asyncId)]: 1527724,
2|bot      | 2022-03-24T00:03:22:         [Symbol(triggerId)]: 1527719
2|bot      | 2022-03-24T00:03:22:       },
2|bot      | 2022-03-24T00:03:22:       [Symbol(kBuffer)]: null,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kBufferCb)]: null,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kBufferGen)]: null,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kCapture)]: false,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kBytesRead)]: 0,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kBytesWritten)]: 0,
2|bot      | 2022-03-24T00:03:22:       [Symbol(connect-options)]: [Object],
2|bot      | 2022-03-24T00:03:22:       [Symbol(RequestTimeout)]: undefined
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     _header: 'POST /v2/private/create-order HTTP/1.1\r\n' +
2|bot      | 2022-03-24T00:03:22:       'Accept: application/json, text/plain, */*\r\n' +
2|bot      | 2022-03-24T00:03:22:       'Content-Type: application/json\r\n' +
2|bot      | 2022-03-24T00:03:22:       'User-Agent: axios/0.21.2\r\n' +
2|bot      | 2022-03-24T00:03:22:       'Content-Length: 285\r\n' +
2|bot      | 2022-03-24T00:03:22:       'Host: api.crypto.com\r\n' +
2|bot      | 2022-03-24T00:03:22:       'Connection: close\r\n' +
2|bot      | 2022-03-24T00:03:22:       '\r\n',
2|bot      | 2022-03-24T00:03:22:     _keepAliveTimeout: 0,
2|bot      | 2022-03-24T00:03:22:     _onPendingData: [Function: noopPendingOutput],
2|bot      | 2022-03-24T00:03:22:     agent: Agent {
2|bot      | 2022-03-24T00:03:22:       _events: [Object: null prototype],
2|bot      | 2022-03-24T00:03:22:       _eventsCount: 2,
2|bot      | 2022-03-24T00:03:22:       _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:       defaultPort: 443,
2|bot      | 2022-03-24T00:03:22:       protocol: 'https:',
2|bot      | 2022-03-24T00:03:22:       options: [Object],
2|bot      | 2022-03-24T00:03:22:       requests: {},
2|bot      | 2022-03-24T00:03:22:       sockets: [Object],
2|bot      | 2022-03-24T00:03:22:       freeSockets: {},
2|bot      | 2022-03-24T00:03:22:       keepAliveMsecs: 1000,
2|bot      | 2022-03-24T00:03:22:       keepAlive: false,
2|bot      | 2022-03-24T00:03:22:       maxSockets: Infinity,
2|bot      | 2022-03-24T00:03:22:       maxFreeSockets: 256,
2|bot      | 2022-03-24T00:03:22:       scheduling: 'lifo',
2|bot      | 2022-03-24T00:03:22:       maxTotalSockets: Infinity,
2|bot      | 2022-03-24T00:03:22:       totalSocketCount: 1,
2|bot      | 2022-03-24T00:03:22:       maxCachedSessions: 100,
2|bot      | 2022-03-24T00:03:22:       _sessionCache: [Object],
2|bot      | 2022-03-24T00:03:22:       [Symbol(kCapture)]: false
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     socketPath: undefined,
2|bot      | 2022-03-24T00:03:22:     method: 'POST',
2|bot      | 2022-03-24T00:03:22:     maxHeaderSize: undefined,
2|bot      | 2022-03-24T00:03:22:     insecureHTTPParser: undefined,
2|bot      | 2022-03-24T00:03:22:     path: '/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:     _ended: true,
2|bot      | 2022-03-24T00:03:22:     res: IncomingMessage {
2|bot      | 2022-03-24T00:03:22:       _readableState: [ReadableState],
2|bot      | 2022-03-24T00:03:22:       _events: [Object: null prototype],
2|bot      | 2022-03-24T00:03:22:       _eventsCount: 3,
2|bot      | 2022-03-24T00:03:22:       _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:       socket: [TLSSocket],
2|bot      | 2022-03-24T00:03:22:       httpVersionMajor: 1,
2|bot      | 2022-03-24T00:03:22:       httpVersionMinor: 1,
2|bot      | 2022-03-24T00:03:22:       httpVersion: '1.1',
2|bot      | 2022-03-24T00:03:22:       complete: true,
2|bot      | 2022-03-24T00:03:22:       headers: [Object],
2|bot      | 2022-03-24T00:03:22:       rawHeaders: [Array],
2|bot      | 2022-03-24T00:03:22:       trailers: {},
2|bot      | 2022-03-24T00:03:22:       rawTrailers: [],
2|bot      | 2022-03-24T00:03:22:       aborted: false,
2|bot      | 2022-03-24T00:03:22:       upgrade: false,
2|bot      | 2022-03-24T00:03:22:       url: '',
2|bot      | 2022-03-24T00:03:22:       method: null,
2|bot      | 2022-03-24T00:03:22:       statusCode: 400,
2|bot      | 2022-03-24T00:03:22:       statusMessage: 'Bad Request',
2|bot      | 2022-03-24T00:03:22:       client: [TLSSocket],
2|bot      | 2022-03-24T00:03:22:       _consuming: false,
2|bot      | 2022-03-24T00:03:22:       _dumped: false,
2|bot      | 2022-03-24T00:03:22:       req: [Circular *1],
2|bot      | 2022-03-24T00:03:22:       responseUrl: 'https://api.crypto.com/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:       redirects: [],
2|bot      | 2022-03-24T00:03:22:       [Symbol(kCapture)]: false,
2|bot      | 2022-03-24T00:03:22:       [Symbol(RequestTimeout)]: undefined
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     aborted: false,
2|bot      | 2022-03-24T00:03:22:     timeoutCb: null,
2|bot      | 2022-03-24T00:03:22:     upgradeOrConnect: false,
2|bot      | 2022-03-24T00:03:22:     parser: null,
2|bot      | 2022-03-24T00:03:22:     maxHeadersCount: null,
2|bot      | 2022-03-24T00:03:22:     reusedSocket: false,
2|bot      | 2022-03-24T00:03:22:     host: 'api.crypto.com',
2|bot      | 2022-03-24T00:03:22:     protocol: 'https:',
2|bot      | 2022-03-24T00:03:22:     _redirectable: Writable {
2|bot      | 2022-03-24T00:03:22:       _writableState: [WritableState],
2|bot      | 2022-03-24T00:03:22:       _events: [Object: null prototype],
2|bot      | 2022-03-24T00:03:22:       _eventsCount: 3,
2|bot      | 2022-03-24T00:03:22:       _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:       _options: [Object],
2|bot      | 2022-03-24T00:03:22:       _ended: true,
2|bot      | 2022-03-24T00:03:22:       _ending: true,
2|bot      | 2022-03-24T00:03:22:       _redirectCount: 0,
2|bot      | 2022-03-24T00:03:22:       _redirects: [],
2|bot      | 2022-03-24T00:03:22:       _requestBodyLength: 285,
2|bot      | 2022-03-24T00:03:22:       _requestBodyBuffers: [],
2|bot      | 2022-03-24T00:03:22:       _onNativeResponse: [Function (anonymous)],
2|bot      | 2022-03-24T00:03:22:       _currentRequest: [Circular *1],
2|bot      | 2022-03-24T00:03:22:       _currentUrl: 'https://api.crypto.com/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:       _timeout: null,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kCapture)]: false
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     [Symbol(kCapture)]: false,
2|bot      | 2022-03-24T00:03:22:     [Symbol(kNeedDrain)]: false,
2|bot      | 2022-03-24T00:03:22:     [Symbol(corked)]: 0,
2|bot      | 2022-03-24T00:03:22:     [Symbol(kOutHeaders)]: [Object: null prototype] {
2|bot      | 2022-03-24T00:03:22:       accept: [Array],
2|bot      | 2022-03-24T00:03:22:       'content-type': [Array],
2|bot      | 2022-03-24T00:03:22:       'user-agent': [Array],
2|bot      | 2022-03-24T00:03:22:       'content-length': [Array],
2|bot      | 2022-03-24T00:03:22:       host: [Array]
2|bot      | 2022-03-24T00:03:22:     }
2|bot      | 2022-03-24T00:03:22:   },
2|bot      | 2022-03-24T00:03:22:   response: {
2|bot      | 2022-03-24T00:03:22:     status: 400,
2|bot      | 2022-03-24T00:03:22:     statusText: 'Bad Request',
2|bot      | 2022-03-24T00:03:22:     headers: {
2|bot      | 2022-03-24T00:03:22:       date: 'Thu, 24 Mar 2022 00:03:22 GMT',
2|bot      | 2022-03-24T00:03:22:       'content-type': 'application/json',
2|bot      | 2022-03-24T00:03:22:       'content-length': '111',
2|bot      | 2022-03-24T00:03:22:       connection: 'close',
2|bot      | 2022-03-24T00:03:22:       'cf-ray': '6f0b24ae9b36c63b-MSP',
2|bot      | 2022-03-24T00:03:22:       'cache-control': 'no-cache, no-store, max-age=0, must-revalidate',
2|bot      | 2022-03-24T00:03:22:       expires: '0',
2|bot      | 2022-03-24T00:03:22:       'strict-transport-security': 'max-age=31536000',
2|bot      | 2022-03-24T00:03:22:       vary: 'Origin, Access-Control-Request-Method, Access-Control-Request-Headers',
2|bot      | 2022-03-24T00:03:22:       'cf-cache-status': 'DYNAMIC',
2|bot      | 2022-03-24T00:03:22:       'content-security-policy': "default-src 'self' https: wss: data: 'unsafe-inline' 'unsafe-eval'",
2|bot      | 2022-03-24T00:03:22:       'expect-ct': 'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"',
2|bot      | 2022-03-24T00:03:22:       pragma: 'no-cache',
2|bot      | 2022-03-24T00:03:22:       'referrer-policy': 'no-referrer',
2|bot      | 2022-03-24T00:03:22:       'x-content-type-options': 'nosniff',
2|bot      | 2022-03-24T00:03:22:       'x-frame-options': 'DENY',
2|bot      | 2022-03-24T00:03:22:       'x-permitted-cross-domain-policies': 'master-only',
2|bot      | 2022-03-24T00:03:22:       'x-xss-protection': '1; mode=block',
2|bot      | 2022-03-24T00:03:22:       'set-cookie': [Array],
2|bot      | 2022-03-24T00:03:22:       server: 'cloudflare'
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     config: {
2|bot      | 2022-03-24T00:03:22:       url: 'https://api.crypto.com/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:       method: 'post',
2|bot      | 2022-03-24T00:03:22:       data: '{"id":2,"method":"private/create-order","params":{"instrument_name":"CVX_USDT","side":"BUY","type":"MARKET","notional":0.119,"client_oid":"mcrbot_invest"},"nonce":1648080201965,"api_key":"MLNpqpT7yRoUihuSCZTdrt","sig":"3f2bc5ee75ab5291c56e4bfa4db31497d4b6115de277f4c3c67f61ee94e5a8ec"}',
2|bot      | 2022-03-24T00:03:22:       headers: [Object],
2|bot      | 2022-03-24T00:03:22:       transformRequest: [Array],
2|bot      | 2022-03-24T00:03:22:       transformResponse: [Array],
2|bot      | 2022-03-24T00:03:22:       timeout: 30000,
2|bot      | 2022-03-24T00:03:22:       adapter: [Function: httpAdapter],
2|bot      | 2022-03-24T00:03:22:       xsrfCookieName: 'XSRF-TOKEN',
2|bot      | 2022-03-24T00:03:22:       xsrfHeaderName: 'X-XSRF-TOKEN',
2|bot      | 2022-03-24T00:03:22:       maxContentLength: -1,
2|bot      | 2022-03-24T00:03:22:       maxBodyLength: -1,
2|bot      | 2022-03-24T00:03:22:       validateStatus: [Function: validateStatus],
2|bot      | 2022-03-24T00:03:22:       transitional: [Object]
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     request: <ref *1> ClientRequest {
2|bot      | 2022-03-24T00:03:22:       _events: [Object: null prototype],
2|bot      | 2022-03-24T00:03:22:       _eventsCount: 7,
2|bot      | 2022-03-24T00:03:22:       _maxListeners: undefined,
2|bot      | 2022-03-24T00:03:22:       outputData: [],
2|bot      | 2022-03-24T00:03:22:       outputSize: 0,
2|bot      | 2022-03-24T00:03:22:       writable: true,
2|bot      | 2022-03-24T00:03:22:       destroyed: false,
2|bot      | 2022-03-24T00:03:22:       _last: true,
2|bot      | 2022-03-24T00:03:22:       chunkedEncoding: false,
2|bot      | 2022-03-24T00:03:22:       shouldKeepAlive: false,
2|bot      | 2022-03-24T00:03:22:       _defaultKeepAlive: true,
2|bot      | 2022-03-24T00:03:22:       useChunkedEncodingByDefault: true,
2|bot      | 2022-03-24T00:03:22:       sendDate: false,
2|bot      | 2022-03-24T00:03:22:       _removedConnection: false,
2|bot      | 2022-03-24T00:03:22:       _removedContLen: false,
2|bot      | 2022-03-24T00:03:22:       _removedTE: false,
2|bot      | 2022-03-24T00:03:22:       _contentLength: null,
2|bot      | 2022-03-24T00:03:22:       _hasBody: true,
2|bot      | 2022-03-24T00:03:22:       _trailer: '',
2|bot      | 2022-03-24T00:03:22:       finished: true,
2|bot      | 2022-03-24T00:03:22:       _headerSent: true,
2|bot      | 2022-03-24T00:03:22:       socket: [TLSSocket],
2|bot      | 2022-03-24T00:03:22:       _header: 'POST /v2/private/create-order HTTP/1.1\r\n' +
2|bot      | 2022-03-24T00:03:22:         'Accept: application/json, text/plain, */*\r\n' +
2|bot      | 2022-03-24T00:03:22:         'Content-Type: application/json\r\n' +
2|bot      | 2022-03-24T00:03:22:         'User-Agent: axios/0.21.2\r\n' +
2|bot      | 2022-03-24T00:03:22:         'Content-Length: 285\r\n' +
2|bot      | 2022-03-24T00:03:22:         'Host: api.crypto.com\r\n' +
2|bot      | 2022-03-24T00:03:22:         'Connection: close\r\n' +
2|bot      | 2022-03-24T00:03:22:         '\r\n',
2|bot      | 2022-03-24T00:03:22:       _keepAliveTimeout: 0,
2|bot      | 2022-03-24T00:03:22:       _onPendingData: [Function: noopPendingOutput],
2|bot      | 2022-03-24T00:03:22:       agent: [Agent],
2|bot      | 2022-03-24T00:03:22:       socketPath: undefined,
2|bot      | 2022-03-24T00:03:22:       method: 'POST',
2|bot      | 2022-03-24T00:03:22:       maxHeaderSize: undefined,
2|bot      | 2022-03-24T00:03:22:       insecureHTTPParser: undefined,
2|bot      | 2022-03-24T00:03:22:       path: '/v2/private/create-order',
2|bot      | 2022-03-24T00:03:22:       _ended: true,
2|bot      | 2022-03-24T00:03:22:       res: [IncomingMessage],
2|bot      | 2022-03-24T00:03:22:       aborted: false,
2|bot      | 2022-03-24T00:03:22:       timeoutCb: null,
2|bot      | 2022-03-24T00:03:22:       upgradeOrConnect: false,
2|bot      | 2022-03-24T00:03:22:       parser: null,
2|bot      | 2022-03-24T00:03:22:       maxHeadersCount: null,
2|bot      | 2022-03-24T00:03:22:       reusedSocket: false,
2|bot      | 2022-03-24T00:03:22:       host: 'api.crypto.com',
2|bot      | 2022-03-24T00:03:22:       protocol: 'https:',
2|bot      | 2022-03-24T00:03:22:       _redirectable: [Writable],
2|bot      | 2022-03-24T00:03:22:       [Symbol(kCapture)]: false,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kNeedDrain)]: false,
2|bot      | 2022-03-24T00:03:22:       [Symbol(corked)]: 0,
2|bot      | 2022-03-24T00:03:22:       [Symbol(kOutHeaders)]: [Object: null prototype]
2|bot      | 2022-03-24T00:03:22:     },
2|bot      | 2022-03-24T00:03:22:     data: {
2|bot      | 2022-03-24T00:03:22:       id: 2,
2|bot      | 2022-03-24T00:03:22:       method: 'private/create-order',
2|bot      | 2022-03-24T00:03:22:       code: 20012,
2|bot      | 2022-03-24T00:03:22:       message: 'ACCOUNT_INSTRUCTION_BANNED_DUE_TO_USER_REGION'
2|bot      | 2022-03-24T00:03:22:     }
2|bot      | 2022-03-24T00:03:22:   },
2|bot      | 2022-03-24T00:03:22:   isAxiosError: true,
2|bot      | 2022-03-24T00:03:22:   toJSON: [Function: toJSON]
2|bot      | 2022-03-24T00:03:22: }

/root/.pm2/logs/bot-out.log last 1000 lines

**Comment by kamigreed** [comment]
Sorry for the long spam. The error popped up today.

**Comment by Thornsnake** [comment]
Thank you, that helped. The error occurs when the bot attempts to create an order for buying CVX.
`"instrument_name":"CVX_USDT","side":"BUY","type":"MARKET","notional":0.119,"client_oid":"mcrbot_invest"`

The error message states: `ACCOUNT_INSTRUCTION_BANNED_DUE_TO_USER_REGION`

It looks like you can not trade CVX because of where you live. The bot fails when it attempts to buy CVX and throws this error. To fix the error, do the following...

1. Stop the bot.
2. Sell all CVX you currently own on that account. (if you have any and can)
3. Put CVX on the exclude list in your config file.
4. Delete the `/data/CoinRemovalList.json` file.
5. Restart the bot.

**Comment by Thornsnake** [comment]
Step 2 and 4 are probably not required, but just in case...

**Comment by kamigreed** [comment]
The weird thing now is I don't see any CVX or its dust in my portfolio and it's still giving this error. I'll go look at my trading logs to see if it really has been sold. 
I've already included it in the config when I submitted this ticket. e.g:
/**
EXCLUDE: ["CVX"],
/**

**Comment by Thornsnake** [comment]
There must be a reason why the bot is trying to buy more CVX. If it's already in the exclude list, then the coin is probably still listet in the `/data/CoinRemovalList.json` file.

**Comment by kamigreed** [comment]
I did what you said (3&4), restarted the bot and checked the CoinRemovalList. No CVX. Then once the time hits 5mins, the error pops up again and CVX gets added back to CoinRemovalList.

I still suspect I have some CVX coins in my crypto acc, but it's just not displaying and that's why it's still giving me this error despite of what you mentioned. [USDC and DOGE was fixed thank you :) ]
 I'll check with crypto.com on this. Thank you.
